src/platform/posix.c

```diff
@@ -556,33 +556,39 @@ bool platform_file_copy(const utf8 *srcPath, const utf8 *dstPath, bool overwrite
 {
 	log_verbose("Copying %s to %s", srcPath, dstPath);
 
-	// If overwrite is not on and the file already exists, return 0
-	if(!overwrite && platform_file_exists(dstPath)) {
-		log_warning("platform_file_copy: Not overwriting %s, because overwrite flag == false", dstPath);
-		return 0;
+	FILE *dstFile;
+
+ 	if (overwrite) {
+		dstFile = fopen(dstPath, "wb");
+	} else {
+		dstFile = fopen(dstPath, "wbx");
 	}
 
+	if (!dstFile) {
+		if (errno == EEXIST) {
+			log_warning("platform_file_copy: Not overwriting %s, because overwrite flag == false", dstPath);
+			return 0;
+		}
 
-	// Open both files and check whether they are opened correctly
-	FILE *srcFile = fopen(srcPath, "r");
-	if(!srcFile) {
-		log_error("Could not open source file %s for copying", srcPath);
+		log_error("Could not open destination file %s for copying", dstPath);
 		return 0;
 	}
 
-	FILE *dstFile = fopen(dstPath, "w");
-	if(!dstFile) {
-		fclose(srcFile);
-		log_error("Could not open destination file %s for copying", dstPath);
+	// Open both files and check whether they are opened correctly
+	FILE *srcFile = fopen(srcPath, "rb");
+	if (!srcFile) {
+		fclose(dstFile);
+		log_error("Could not open source file %s for copying", srcPath);
 		return 0;
 	}
 
 	size_t amount_read = 0;
 
 	char* buffer = (char*) malloc(FILE_BUFFER_SIZE);
-	while(amount_read = fread(buffer, FILE_BUFFER_SIZE, 1, srcFile)) {
+	while ((amount_read = fread(buffer, FILE_BUFFER_SIZE, 1, srcFile))) {
```

Comment:
The extra parenthesis here will prevent clang from giving a warning about using assignment instead of equality  comparison. It's best practice to this when assigning and equality comparing in the same expression.

```text
[ 34%] Building C object CMakeFiles/openrct2.dir/src/platform/posix.c.o
src/platform/posix.c:588:20: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
        while(amount_read = fread(buffer, FILE_BUFFER_SIZE, 1, srcFile)) {
              ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/platform/posix.c:588:20: note: place parentheses around the assignment to silence this warning
        while(amount_read = fread(buffer, FILE_BUFFER_SIZE, 1, srcFile)) {
                          ^
              (                                                        )
src/platform/posix.c:588:20: note: use '==' to turn this assignment into an equality comparison
        while(amount_read = fread(buffer, FILE_BUFFER_SIZE, 1, srcFile)) {
                          ^
                          ==
1 warning generated.
```